lib/services/local/dynamic_access.go

```diff
@@ -51,33 +52,43 @@ func (s *DynamicAccessService) CreateAccessRequest(ctx context.Context, req serv
 }
 
 func (s *DynamicAccessService) SetAccessRequestState(ctx context.Context, name string, state services.RequestState) error {
-	item, err := s.Get(ctx, accessRequestKey(name))
-	if err != nil {
-		if trace.IsNotFound(err) {
-			return trace.NotFound("cannot set state of access request %q (not found)", name)
+	// Setting state is attempted multiple times in the event of concurrent writes.
+	// The reason we bother to re-attempt is because state updates aren't meant
+	// to be "first come first serve".  Denials should overwrite approvals, but
+	// approvals should not overwrite denials.
+	for i := 0; i < maxCmpAttempts; i++ {
```

Comment:
I haven't spotted any delay, randomization or backoff here? This will create a problem in production. Please use utils.Linear with some backoff